src/Classifier.php

```diff
@@ -13,33 +13,341 @@
 class Classifier
 {
     /**
-     * @var array
+     * @var array Array of regular expressions which match mobile devices
      */
-    public $mobileOSs = ['windows phone 6.5','windows ce','symbian os'];
+    public $mobileDevices = [
```

Comment:
Is it deliberate that these properties are public? Could we make them private (or protected if we wanted the class to be extensible through inheritance?)

If these become private, then we can make the assumption that the `rules` property is a direct function of these properties, and since the properties never change, we could consider computing the rules once statically against the class rather than each time an instance of Classifier is constructed.

Reply:
How are you envisaging computing them against the class? Perhaps they could be stored in a file?

Reply:
I was thinking something along the lines of - convert all these instance properties to static properties, add a `getRules` method which populates the `rules` array on first call and then just returns it on subsequent calls.

All of this is potentially overkill if we assume that there's only likely to be a single instance of classifier created and set up, but it's worth considering at least.

But first off, these properties don't need to be public unless we actually want to allow them to be accessed externally to this class.